lib/linguist/heuristics.yml

```diff
@@ -567,6 +567,18 @@ disambiguations:
   rules:
   - language: XML
     pattern: '<TS\b'
+  - language: AssemblyScript
+    negative_pattern: 
+    - '\s+undefined\s+'
+    - '\s+any\s+'
+    - '\s+unknown\s+'
+    pattern:
+    - '@inline'
+    - '@final'
+    - '@unmanaged'
+    - '// @ts-ignore: decorator'
+    - '[iuf](8|16|32|64)'
+    - 'usize'
```

Comment:
```suggestion
    - '\b[iu]size'
```

Reply:
why only `\b` on the beginning and not the end?

Reply:
because we should accept `isize` or `usize` but not `iusize` or `uisize`

Reply:
yes, that's the `\b` at the front, but thy not `\b[iu]size\b`?

Reply:
You could add this at the end but usually it's unnecessary. But it's depend on how tokens match - per lines or per words.

Reply:
It doesn't match tokens. The regex is matched against the whole file as far as I could figure out.

Reply:
well, in this case make sense wrap tokens to `\b`

Reply:
The heuristics are only run against the first bytes of the file:

https://github.com/github/linguist/blob/e2f638d17598e84f6f91c2d03c2f3e27f22a2163/lib/linguist/heuristics.rb#L6

Reply:
`\b[iu]size` is still going to match stuff like `$isize` and `foo.usize`.

Reply:
So something like `[\s:<][iu]size[\s>)]` would be more accurate.
(The `:` is for variable defintions like `const ptr:usize`, the `<>` for generics like `Array<usize>` and the `)` for `(ptr: usize)=>void`)

Reply:
that's true. It could be false positive. But at the same time this should be also valid: `:isize` for example `function foo():isize`.

Reply:
All possible variants: `Array<usize>`, `()=>usize`, `:isize`, `isize(x)`, `<isize>x`

Reply:
It seems like a more appropriate border would be `[^.\w]`, so it doesn't match inside other words and not as a property.